pkg/client/nodes_test.go

```diff
@@ -38,10 +41,41 @@ func TestListMinions(t *testing.T) {
 		},
 		Response: Response{StatusCode: 200, Body: &api.NodeList{ListMeta: api.ListMeta{ResourceVersion: "1"}}},
 	}
-	response, err := c.Setup().Nodes().List()
+	response, err := c.Setup().Nodes().List(labels.Everything())
 	c.Validate(t, response, err)
 }
 
+func TestListMinionsLabels(t *testing.T) {
+	ns := api.NamespaceNone
+	labelSelectorQueryParamName := api.LabelSelectorQueryParam(testapi.Version())
+	c := &testClient{
+		Request: testRequest{
+			Method: "GET",
+			Path:   testapi.ResourcePath(getNodesResourceName(), "", ""),
+			Query:  buildQueryValues(ns, url.Values{labelSelectorQueryParamName: []string{"foo=bar,name=baz"}})},
+		Response: Response{
+			StatusCode: 200,
+			Body: &api.NodeList{
+				Items: []api.Node{
+					{
+						ObjectMeta: api.ObjectMeta{
+							Labels: map[string]string{
+								"foo":  "bar",
+								"name": "baz",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	c.Setup()
+	c.QueryValidator[labelSelectorQueryParamName] = validateLabels
+	selector := labels.Set{"foo": "bar", "name": "baz"}.AsSelector()
+	receivedPodList, err := c.Nodes().List(selector)
```

Comment:
Sorry I'm being dense, but I still don't understand. Is the List() that's invoked here the one in fake_minions.go? If so, where does the list of nodes that gets returned get populated?